src/ggl-setup.py

```diff
@@ -0,0 +1,524 @@
+# HELPER FUNCTIONS
+# ===============================================
+def _download_source(commit_id: str, max_retries=3) -> bool:
+    url = f"https://github.com/aws-greengrass/aws-greengrass-lite/archive/{commit_id}.zip"
+    src_path = "./aws-greengrass-lite.zip"
+    extracted_folder = f"aws-greengrass-lite-{commit_id}"
+    target_folder = "aws-greengrass-lite"
+
+    if os.path.isdir(target_folder):
+        print("Please clean up before downloading the aws-greengrass-lite")
+        return False
+
+    for attempt in range(max_retries):
+        try:
+            response = requests.get(url)
+
+            if response.status_code != 200:
+                print(
+                    f"Failed to download aws-greengrass-lite, status code: {response.status_code}"
+                )
+                if attempt < max_retries - 1:
+                    continue
+                return False
+
+            with open(src_path, 'wb') as f:
+                f.write(response.content)
+
+            if _unzip_file(src_path, '.'):
+                os.rename(extracted_folder, target_folder)
+                print("Successfully downloaded aws-greengrass-lite")
+                return True
+
+        except requests.exceptions.RequestException as e:
+            print(f"Error when downloading aws-greengrass-lite, {str(e)}")
+            if attemp == max_retries - 1:
+                print(
+                    f"Failed to download aws-greengrass-lite after {max_retries} attempts"
+                )
+                return False
+
+
+def _unzip_file(src_path: str, dest_path: str) -> bool:
+    try:
+        with zipfile.ZipFile(src_path, 'r') as zip_ref:
+            zip_ref.extractall(dest_path)
+
+        os.remove(src_path)
+        return True
+
+    except Exception as e:
+        print(f"Error during extraction: {str(e)}")
+        return False
+
+
+def _install_build_dependencies() -> bool:
+    try:
+        # List of packages to install
+        packages = [
+            "build-essential", "pkg-config", "cmake", "git", "curl",
+            "libssl-dev", "libcurl4-openssl-dev", "uuid-dev", "libzip-dev",
+            "libsqlite3-dev", "libyaml-dev", "libsystemd-dev", "libevent-dev",
+            "liburiparser-dev", "cgroup-tools"
+        ]
+
+        # Update package list
+        update_command = ["sudo", "apt", "update"]
+        subprocess.run(update_command, check=True)
+
+        # Install packages
+        install_command = ["sudo", "apt", "install", "-y"] + packages
+        subprocess.run(install_command, check=True)
+
+        print("Successfully updated and installed dependencies")
+        return True
+
+    except Exception as e:
+        print(f"Error when installing dependencies: {e}", file=sys.stderr)
+        return False
+
+
+def _add_user_and_group(user: str, group: str):
+    try:
+        # Create the group
+        if not _check_group_exists(group):
+            subprocess.run(['sudo', 'groupadd', group], check=True)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method is running with elevated privileges. Privilege elevation occurs when you access resources with an extended set of permissions. When unattended, these privileges might allow attackers to compromise the system. Do not use elevated privileges when they can be avoided.

[Learn more](https://cwe.mitre.org/data/definitions/269.html)

Similar issue at line number 238.